### ISSUE TITLE
Get-NavContainerAppInfo  fails with recursive loop

### ISSUE THREAD

**Issue by kine** [microsoft/navcontainerhelper#562]
I am running this command on my container:
`$dockerapp = Get-NavContainerAppInfo -containerName $ContainerName -tenantSpecificProperties`

And result is this error because recursive call between AddDependencies, AddAnApp and AddDependency:

```
The script failed due to call depth overflow.
at AddAnApp, <No file>: line 6
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at AddDependency, <No file>: line 15
at <ScriptBlock>, <No file>: line 19
at AddDependencies, <No file>: line 19
at AddAnApp, <No file>: line 8
at <ScriptBlock>, <No file>: line 27
at <ScriptBlock>, <No file>: line 27
The script failed due to call depth overflow.
At C:\Users\adm_ksacek\Documents\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.93\ContainerHandling\Invoke-Script
nNavContainer.ps1:37 char:13
+             Invoke-Command -Session $session -ScriptBlock $scriptbloc ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (0:Int32) [ForEach-Object], RuntimeException
    + FullyQualifiedErrorId : CallDepthOverflow,Microsoft.PowerShell.Commands.ForEachObjectCommand
    + PSComputerName        : 0b0e765c0e959938653f192b6721ad34e5420c2b90d9ac83c60e702319b4f43d
```
module version: 0.6.2.93

**Comment by freddydk** [comment]
what image?
which apps are installed?

**Comment by kine** [comment]
Just working on the analysis. It is on our QA server where all our Apps are installed. Image is bcinsider.azurecr.io/bcsandbox (some two month old version):
                "osversion": "10.0.14393.2969",
                "platform": "14.0.32341.0",
                "tag": "0.0.9.7",
                "version": "14.1.32447.0"

But the functionality worked two days ago, before you changed the function in commit e7710a1241f5e8bcf77d9141a6261abf87b137cf.

I general, we have few apps which have dependency on another app. There could be 3 levels (base app, app, test app) dependencies. I am going to debug the script...

**Comment by freddydk** [comment]
if you set Sort to None (should have been default) - then you will circumvent the sorting.
If you can do this - and email me the output - I would very much like to figure out what's failing.

**Comment by kine** [comment]
Ok, seems the problem is that default is empty and not None. if I set sort parameter, it works.

**Comment by kine** [comment]
If I use 
`$dockerapp = Get-NavContainerAppInfo -containerName DOCKERBC -tenantSpecificProperties -sort DependenciesFirst`

It fails. I will send more info later, after I will dig into it more...

**Comment by freddydk** [comment]
It will probably fail with DependenciesFirst then - I would like to know why?

**Comment by kine** [comment]
Email on the way.

**Comment by freddydk** [comment]
Fix shipped in 0.6.2.94
